src/app/clusters/webrtc-transport-requestor-server/webrtc-transport-requestor-cluster.cpp

```diff
@@ -0,0 +1,355 @@
+            return Protocols::InteractionModel::Status::InvalidCommand;
+        }
+    }
+
+    // Convert ICETransportPolicy from CharSpan to std::string.
+    if (req.ICETransportPolicy.HasValue())
+    {
+        args.iceTransportPolicy.SetValue(std::string(req.ICETransportPolicy.Value().data(), req.ICETransportPolicy.Value().size()));
+    }
+
+    // Delegate processing: handle the SDP offer, gather ICE candidates, SDP answer, etc.
+    return mDelegate.HandleOffer(sessionId, args);
+}
+
+DataModel::ActionReturnStatus WebRTCTransportRequestorServer::HandleAnswer(const Commands::Answer::DecodableType & req,
+                                                                           const CommandHandler & commandHandler)
+{
+    uint16_t sessionId = req.webRTCSessionID;
+    auto sdpSpan       = req.sdp;
+
+    if (!IsPeerNodeSessionValid(sessionId, commandHandler))
+    {
+        return Protocols::InteractionModel::Status::NotFound;
+    }
+
+    std::string sdpAnswer(sdpSpan.data(), sdpSpan.size());
+
+    // Delegate handles Answer command received.
+    return mDelegate.HandleAnswer(sessionId, sdpAnswer);
+}
+
+DataModel::ActionReturnStatus
+WebRTCTransportRequestorServer::HandleICECandidates(const Commands::ICECandidates::DecodableType & req,
+                                                    const CommandHandler & commandHandler)
+{
+    uint16_t sessionId = req.webRTCSessionID;
+
+    std::vector<ICECandidateStruct> candidates;
+    auto iter = req.ICECandidates.begin();
+    while (iter.Next())
+    {
+        // Get current candidate.
+        const ICECandidateStruct & candidate = iter.GetValue();
+        candidates.push_back(candidate);
+    }
+
+    // Check the validity of the list.
+    CHIP_ERROR err = iter.GetStatus();
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "HandleICECandidates: ICECandidates list error: %" CHIP_ERROR_FORMAT, err.Format());
+        return Protocols::InteractionModel::Status::InvalidCommand;
+    }
+
+    if (candidates.empty())
+    {
+        ChipLogError(Zcl, "HandleICECandidates: No ICE candidates provided.");
+        return Protocols::InteractionModel::Status::ConstraintError;
+    }
+
+    if (!IsPeerNodeSessionValid(sessionId, commandHandler))
+    {
+        return Protocols::InteractionModel::Status::NotFound;
+    }
+
+    return mDelegate.HandleICECandidates(sessionId, candidates);
+}
+
+DataModel::ActionReturnStatus WebRTCTransportRequestorServer::HandleEnd(const Commands::End::DecodableType & req,
+                                                                        const CommandHandler & commandHandler)
+{
+    uint16_t sessionId = req.webRTCSessionID;
+    auto reason        = req.reason;
+
+    // Validate the reason field against the allowed enum values.
+    if (reason == WebRTCEndReasonEnum::kUnknownEnumValue)
+    {
+        ChipLogError(Zcl, "HandleEnd: Invalid reason value %u.", static_cast<uint8_t>(reason));
```

Comment:
```suggestion
        ChipLogError(Zcl, "HandleEnd: Invalid reason value %u.", to_underlying(reason));
```
I know, pre-existing....